src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrLength = this.length;
+    let str = '';
+
+    for (let i = 0; i < arrLength; i++) {
+      if (typeof this[i] === 'undefined' || this[i] === null) {
+        this[i] = '';
+      }
+
+      if (i !== arrLength - 1) {
+        str += this[i] + String(separator);
```

Comment:
```suggestion
        str += this[i] + separator;
```

Reply:
in this case test throw me an error

https://prnt.sc/coQLsyVk6oga